fastplotlib/tools/_cursor.py

```diff
@@ -0,0 +1,357 @@
+
+        size_space: "screen" | "world", default "screen"
+            size space of the cursor, if "screen" the ``size`` is exact screen pixels.
+            if "world" the ``size`` is in world-space
+
+        """
+
+        self._cursors: dict[Subplot, [pygfx.Points | pygfx.Group[pygfx.Line]]] = dict()
+
+        self._mode = None
+        self.mode = mode
+        self.size = size
+        self.color = color
+        self.marker = marker
+        self.edge_color = edge_color
+        self.edge_width = edge_width
+        self.alpha = alpha
+        self.size_space = size_space
+
+        self._pause = False
+
+        self._position = [0, 0]
+
+    @property
+    def mode(self) -> Literal["crosshair", "marker"]:
+        return self._mode
+
+    @mode.setter
+    def mode(self, mode: Literal["crosshair", "marker"]):
+        if not (mode == "crosshair" or mode == "marker"):
+            raise ValueError(
+                f"mode must be one of: 'crosshair' | 'marker', you passed: {mode}"
+            )
+
+        if mode == self.mode:
+            return
+
+        # mode has changed, clear and create new world objects
+        subplots = list(self._cursors.keys())
+
+        self.clear()
+
+        for subplot in subplots:
+            self.add_subplot(subplot)
+
+        self._mode = mode
+
+    @property
+    def size(self) -> float:
+        return self._size
+
+    @size.setter
+    def size(self, new_size: float):
+        for c in self._cursors.values():
+            if self.mode == "marker":
+                c.material.size = new_size
+            elif self.mode == "crosshair":
+                h, v = c.children
+                h.material.thickness = new_size
+                v.material.thickness = new_size
+
+        self._size = new_size
+
+    @property
+    def size_space(self) -> Literal["screen", "world"]:
+        return self._size_space
+
+    @size_space.setter
+    def size_space(self, space: Literal["screen", "world"]):
+        if space not in ["screen", "world", "model"]:
+            raise ValueError(
+                f"valid `size_space` is one of: 'screen' | 'world'. You passed: {space}"
+            )
+
+        for c in self._cursors.values():
+            if self.mode == "marker":
+                c.material.size_space = space
+
+            elif self.mode == "crosshair":
+                h, v = c.children
+                h.material.thickness_space = space
+                v.material.thickness_space = space
+
+        self._size_space = space
+
+    @property
+    def color(self) -> pygfx.Color:
+        return self._color
+
+    @color.setter
+    def color(self, new_color):
+        new_color = pygfx.Color(new_color)
+
+        for c in self._cursors.values():
+            c.material.color = new_color
+
+        self._color = new_color
+
+    @property
+    def marker(self) -> str:
+        return self._marker
+
+    @marker.setter
+    def marker(self, new_marker: str):
```

Comment:
should validate the `new_marker` with valid markers (similar to what scatter does)